Add documentation for setting proxy (#252)
AxisCommunications/docker-acap: README.md

```

#### Proxy Setup

If the device is located behind a corporate proxy you may need to set the correct environment variables. This is done by configuring proxy behavior for dockerd in the daemon.json file as described in ['Configure the Docker daemon to use a proxy server'][docker-proxy].

To do this for the application you can either use [developer mode][developermode] or build the application manually with a pre-created daemon.json that is copied onto the device by the postinstall script.

> daemon.json file should be located at /usr/local/packages/dockerdwrapper/localdata/daemon.json on the device and should include the fallowing properties.

Reviewer: ```suggestion
The daemon.json file should be located at `/usr/local/packages/dockerdwrapper/localdata/daemon.json` on the device and should include the following properties.
```

```json
{
  "proxies": {

>     "http-proxy": "http://<myproxy>:<port>",

Reviewer: use the same example here as on the docs.docker.com page

    "https-proxy": "http://<myproxy>:<port>",
    "no-proxy": "localhost,127.0.0.0/8,10.0.0.0/8,192.168.0.0/16,172.16.0.0/12,.<domain>"
 }
}
```

> If [developer mode][developermode] is used, after downloading the application SSH user acap-dockerdwrapper will be created. Update the password of acap-dockerdwrapper. SSH user acap-dockerdwrapper can be used to write in the daemon.json file, application has to be restarted once daemon.json is updated.

Reviewer: Setting the contents of the daemon.json file can be done either by adding it to the source code and rebuilding the application or by ssh:ing into the device with an already installed application. In the latter case [developer mode][developermode] is needed, see that documentation for further details. Also note that, if the application is running when the file is updated, it needs to be restarted for the change to take effect.

```

> #### Proxy Setup

Reviewer: I think you should add a few lines on why someone would need to set the proxy env variables. e.g.
If the device is located behind a corporate proxy you may need to set the correct environment variables. This is done by configuring proxy behavior for dockerd in the daemon.json file as described in [Configure the Docker daemon to use a proxy server](https://docs.docker.com/config/daemon/systemd/#httphttps-proxy)][https://docs.docker.com/config/daemon/systemd/#httphttps-proxy].

To do this for the application you can either use [developer mode](#developermode) or build the application manually with a pre-creaeted daemon.json that is copied onto the device by the postinstall script 
^ The last section should be extended to be a bit more detailed in both cases

I do not think you should add the shell script - only describe what the daemon.json should contain, where it is located on the device and that the application needs to be restarted once it is updated.

Author: Removed the script and the note

> [!NOTE]
> **From AXIS OS 12.0**
>

> > To use the shell script provided to set proxy, [developer mode](#developermode) certificate has to be set on camera, to be able use acap-dockerdwrapper ssh user instead of root ssh user.

Reviewer 2: SSH as root requires another (Axis internal) certificate, so rephrase it something like: "... developer mode certificate has to be installed on the camera (device? which term is used), in order to use SSH."

Will the script take care of which username to use? I suggest that it defaults to acap-dockerdwrapper and only uses root if a special option "--root" or similar is passed in.

```

#### Proxy Setup

> If the device is located behind a corporate proxy you may need to set the correct environment variables.

Reviewer 3: In the Docker documentation there are two ways to set a proxy,

- configuring the daemon
- setting environment variables

These instructions configures the daemon, so it seems a bit strange to mention environment variables.

Maybe something like this?
```suggestion
If the device is located behind a proxy the Docker daemon needs to be configured.
```

This is done by configuring proxy behavior for dockerd in the daemon.json file as described in
['Configure the Docker daemon to use a proxy server'][docker-proxy].

The daemon.json file should be located at `/usr/local/packages/dockerdwrapper/localdata/daemon.json`
on the device and should include the following properties:

```json
{
  "proxies": {
    "http-proxy": "http://proxy.example.com:3128",
    "https-proxy": "https://proxy.example.com:3129",
    "no-proxy": "*.test.example.com,.example.org,127.0.0.0/8"
  }
}
```

Setting the contents of the daemon.json file can be done either by adding it to the source code and

> rebuilding the application or by ssh:ing into the device with an already installed application and updating

Reviewer 3: I think this sounds a bit better than "ssh:ing"
Reference: https://www.redhat.com/sysadmin/access-remote-systems-ssh 

```suggestion
rebuilding the application or by logging into the device over SSH with an already installed application and updating
```

the file.

> In the latter case [developer mode][developermode] is needed, see that documentation for further details.

Reviewer 3: In acap-documentation we have chosen to use Developer Mode over developer mode.
```suggestion
In the latter case [Developer Mode][developermode] is needed, see that documentation for further details.
```

 }
}
```

> If [developer mode][developermode] is used, after downloading the application SSH user acap-dockerdwrapper will be created. Update the password of acap-dockerdwrapper. SSH user acap-dockerdwrapper can be used to write in the daemon.json file, application has to be restarted once daemon.json is updated.

Reviewer 2: ```suggestion
If [developer mode][developermode] is used, the SSH user acap-dockerdwrapper will be created when the application is installed. Once the password of acap-dockerdwrapper has been set, SSH can be used to edit daemon.json file. The application has to be restarted in order for changes in daemon.json to take effect.
```

Author: This part has been changes
